src/main/java/org/ow2/proactive/catalog/dto/CatalogObjectMetadata.java

```diff
@@ -95,11 +99,12 @@ public CatalogObjectMetadata(CatalogObjectRevisionEntity catalogObject) {
              catalogObject.getCatalogObject().getContentType(),
              catalogObject.getCommitTime(),
              catalogObject.getCommitMessage(),
-             KeyValueEntityToDtoTransformer.to(catalogObject.getKeyValueMetadataList()));
+             KeyValueEntityToDtoTransformer.to(catalogObject.getKeyValueMetadataList()),
+             catalogObject.getCatalogObject().getExtension());
     }
 
     public CatalogObjectMetadata(String bucketName, String name, String kind, String contentType, long commitTime,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)